sdk/include/hipdnn_sdk/test_utilities/FlatbufferDatatypeMapping.hpp

```diff
@@ -0,0 +1,100 @@
+// Copyright © Advanced Micro Devices, Inc., or its affiliates.
+// SPDX-License-Identifier: MIT
+
+#pragma once
+
+#include <hipdnn_sdk/data_objects/data_types_generated.h>
+#include <hipdnn_sdk/utilities/UtilsBfp16.hpp>
+#include <hipdnn_sdk/utilities/UtilsFp16.hpp>
+
+namespace
+{
+template <typename T>
+struct TypeWrapper
+{
+    using type = T;
+};
+
+template <hipdnn_sdk::data_objects::DataType DT>
+constexpr auto getTypeWrapper()
+{
+    using DataType = hipdnn_sdk::data_objects::DataType;
+
+    if constexpr(DT == DataType::FLOAT)
+    {
+        return TypeWrapper<float>{};
+    }
+    else if constexpr(DT == DataType::HALF)
+    {
+        return TypeWrapper<half>{};
+    }
+    else if constexpr(DT == DataType::DOUBLE)
+    {
+        return TypeWrapper<double>{};
+    }
+    else if constexpr(DT == DataType::INT32)
+    {
+        return TypeWrapper<int32_t>{};
+    }
+    else if constexpr(DT == DataType::BFLOAT16)
+    {
+        return TypeWrapper<hip_bfloat16>{};
+    }
+    else
+    {
+        static_assert(DT != DT, "Unsupported DataType");
+    }
+}
+
+template <typename T>
+constexpr auto toDataType()
+{
+    if constexpr(std::is_same_v<T, float>)
+    {
+        return hipdnn_sdk::data_objects::DataType::FLOAT;
+    }
+    else if constexpr(std::is_same_v<T, half>)
+    {
+        return hipdnn_sdk::data_objects::DataType::HALF;
+    }
+    else if constexpr(std::is_same_v<T, double>)
+    {
+        return hipdnn_sdk::data_objects::DataType::DOUBLE;
+    }
+    else if constexpr(std::is_same_v<T, int32_t>)
+    {
+        return hipdnn_sdk::data_objects::DataType::INT32;
+    }
+    else if constexpr(std::is_same_v<T, hip_bfloat16>)
+    {
+        return hipdnn_sdk::data_objects::DataType::BFLOAT16;
+    }
+    else
+    {
+        static_assert(sizeof(T) == 0, "Unsupported native type");
+    }
+}
+
+}
+
+namespace hipdnn_sdk
```

Comment:
This is kind of verbose, especially since you put the braces on the next line.

What about:
namespace hipdnn_sdk::test_utilities
{